server/apps/immich/src/api-v1/album/album.service.ts

```diff
@@ -174,4 +196,40 @@ export class AlbumService {
       album.albumThumbnailAssetId = dto.albumThumbnailAssetId || null;
     }
   }
+
+  async createAlbumSharedLink(authUser: AuthUserDto, dto: CreateAlbumShareLinkDto): Promise<SharedLinkResponseDto> {
+    const album = await this._getAlbum({ authUser, albumId: dto.albumId });
+
+    const sharedLink = await this.shareCore.createSharedLink(authUser.id, {
+      sharedType: SharedLinkType.ALBUM,
+      expiredAt: dto.expiredAt,
+      allowUpload: dto.allowUpload,
+      album: album,
+      assets: album.assets?.map((asset) => asset.assetInfo) || [],
+      description: dto.description,
+    });
+
+    return mapSharedLinkToResponseDto(sharedLink);
+  }
+
+  async updateAssetInSharedLinks(album: AlbumEntity) {
+    // Update asset in shared links
+    if (album.sharedLinks.length > 0) {
+      const sharedLinkIds = album.sharedLinks.map((link) => link.id);
+      if (album.assets && album.assets.length > 0) {
+        const assets = album.assets.map((asset) => asset.assetInfo);
+
+        for (const sharedLinkId of sharedLinkIds) {
+          await this.shareCore.updataAssetsInSharedLink(sharedLinkId.toString('hex'), assets);
+        }
+      }
+    }
+  }
```

Comment:
I think all this code can go away if we instead update our queries to load assets for the linked album instead.